typescript/frontend-cicd/lib/frontend-cicd-stack.ts

```diff
@@ -0,0 +1,121 @@
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as codecommit from 'aws-cdk-lib/aws-codecommit';
+import * as codebuild from 'aws-cdk-lib/aws-codebuild';
+import * as codepipeline from 'aws-cdk-lib/aws-codepipeline';
+import * as codepipelineActions from 'aws-cdk-lib/aws-codepipeline-actions';
+import { Construct } from 'constructs';
+
+export class FrontendCicdStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    // The code that defines your stack goes here
+    const pipeLine = new codepipeline.Pipeline(this, 'MyFrontEndPipeline', {
+      pipelineName: 'MyFrontEndPipeline'
+    });
+
+    // Source
+    const repository = new codecommit.Repository(this, 'MyFrontEndRepo', {
+      repositoryName: 'MyFrontEndRepo',
+      description: 'My Frontend Repository'
+    });
+
+    const sourceOutput = new codepipeline.Artifact();
+    const sourceAction = new codepipelineActions.CodeCommitSourceAction({
+      actionName: 'CodeCommit',
+      repository: repository,
+      branch: 'master',   //or main
+      output: sourceOutput
+    });
+    pipeLine.addStage({
+      stageName: 'Source',
+      actions: [sourceAction],
+    });
+    //
+
+    // Build
+    const project = new codebuild.Project(this, 'MyFrontEndBuild', {
+      source: codebuild.Source.codeCommit({ repository }),
+      buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yaml'),
+      environment: {
+        buildImage: codebuild.LinuxBuildImage.STANDARD_5_0,
+        privileged: true
+      },
+      environmentVariables: {
+        tag: {
+          value: 'cdk'
+        }
+      }
+    });
+
+    const buildOutput = new codepipeline.Artifact();
+    const buildAction = new codepipelineActions.CodeBuildAction({
+      actionName: 'CodeBuild',
+      project: project,
+      input: sourceOutput,
+      outputs: [buildOutput],
+      // executeBatchBuild: true,
+      // combineBatchBuildArtifacts: true
+    });
+    pipeLine.addStage({
+      stageName: 'Build',
+      actions: [buildAction],
+    });
+
+    // Change bucket name global unique.
+    const bucket = new s3.Bucket(this, '<YOUR UNIQUE BUCKET NAME>', {
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      blockPublicAccess: new s3.BlockPublicAccess({
+        blockPublicPolicy: true
+      }),
+      removalPolicy: RemovalPolicy.DESTROY, //Auto delete bucket when CFN destroyed.
+      autoDeleteObjects: true,              //Auto delete objects when CFN destroyed.
+      objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED
+    });
+    bucket.addToResourcePolicy(new iam.PolicyStatement({
+      actions: [
+        's3:*'
+      ],
+      resources: [bucket.arnForObjects('*')],
+      principals: [new iam.AccountRootPrincipal()]
+    }));
```

Comment:
This isn't the cleanest way to do this, but it's A Way.

you should prefer using the `grantX` pattern instead of creating inline resource policies.

You should call this out in the README that the root account principal is being given permission for all s3 actions on all buckets.

Reply:
I purposed to add this policy to make customer refer their bucket in their account as well. As your request, I have changed it not to use RootAccount, but to use IAM User account. Please check.